sdk/search/Azure.Search.Documents/src/Models/SearchQueryType.Serialization.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable disable
+
+using System;
+
+namespace Azure.Search.Documents.Models
+{
+    /// <summary>
+    /// TODO: Remove this custom serializer when enabling Semantic search
+    /// </summary>
+    internal static partial class SearchQueryTypeExtensions
+    {
+        public static string ToSerialString(this SearchQueryType value) => value switch
```

Comment:
Why not just declare these as regular methods on the partial `SearchQueryType` struct?

Reply:
`SearchQueryType` [is an enum](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/search/Azure.Search.Documents/src/Models/SearchQueryType.cs).

Reply:
Oh...wow! It's been a while since I've seen an actual `enum` in our code! I just assumed it was an enum-like struct! 😄